### PR TITLE
CB Response support events results

### PR DESCRIPTION
**Pull request by barvhaim**

### CODE REVIEW

File: stix_shifter_modules/carbonblack/stix_translation/results_translator.py [hunk 1]
```diff
@@ -1,11 +1,19 @@
...
 import json
 
 
 class ResultsTranslator(BaseResultTranslator):
 
+    def __init__(self, options, dialect, base_file_path=None, callback=None):
+        super().__init__(options, dialect, base_file_path, callback)
+        show_events = Connector.get_show_events_mode({'options': options})
+        if show_events:
+            filepath = os.path.abspath(os.path.join(base_file_path, "json", "to_stix_map_events.json"))
```
Comment (aviv1ron1) [inline stix_shifter_modules/carbonblack/stix_translation/results_translator.py:14]: we need to make sure that the original mappings have the same results as the new ones.
ideally we would load both configs - and not define the process object in both files

File: stix_shifter_modules/carbonblack/stix_transmission/api_client.py [hunk 2]
```diff
@@ -6,7 +6,8 @@ class APIClient():
     PROCESS_ENDPOINT = 'process'
 
     def __init__(self, connection, configuration):
-        self.endpoint_start = 'api/v1/'
+        self.endpoint_start_v1 = 'api/v1/'
```
Comment (aviv1ron1) [inline stix_shifter_modules/carbonblack/stix_transmission/api_client.py:9]: add comment to explain why V1 and V4 are different

File: stix_shifter_modules/carbonblack/stix_transmission/api_client.py [hunk 3]
```diff
@@ -19,15 +20,17 @@ def __init__(self, connection, configuration):
...
-        data = [("q", query_expression), ("start", start), ("rows", rows)]
-        sort_by = 'start asc' # The purpose of this is to maintain order stability when doing paging
+        event_endpoint = self.endpoint_start_v4 + self.PROCESS_ENDPOINT + '/{}/{}/event'.format(process_id, segment_id)
+        data = []
+        return self.client.call_api(event_endpoint, 'GET', headers, urldata=data, timeout=self.timeout)
 
-        data.append(("sort", sort_by))
-
-        return self.client.call_api(endpoint, 'GET', headers, urldata=data, timeout=self.timeout)
+    def run_processes_search(self, query_expression, start=0, rows=10):
```
Comment (aviv1ron1) [inline stix_shifter_modules/carbonblack/stix_transmission/api_client.py:32]: fix naming convention and add comments to explain

File: stix_shifter_modules/carbonblack/stix_transmission/event_parser.py [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+crossproc_fields = ['crossproc_type', 'event_time', 'target_unique_id', 'target_md5', 'target_path', 'sub_type',
+                    'requested_access', 'tamper']
+str_event_fields = {
+    'regmod': regmod_fields,
+    'filemod': filemod_fields,
+    'modload': modload_fields,
+    'crossproc': crossproc_fields
+}
+CB_PROVIDER = 'Carbon Black Response'
+filemod_operation_dict = {'1': 'Created the file', '2': 'First wrote to the file', '4': 'Deleted the file', '8': 'Last wrote to the file'}
```
Comment (aviv1ron1) [inline stix_shifter_modules/carbonblack/stix_transmission/event_parser.py:27]: add in comments the url of the documentation from where this was taken

File: stix_shifter_modules/carbonblack/stix_transmission/event_parser.py [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+                     'parent_name': cbr_process.get('parent_name'),
+                     'process_cmdline': cbr_process.get('cmdline'),
+                     'provider': CB_PROVIDER
+                     }
+    ip4_val = cbr_process.get('comms_ip')
+    common_fields['device_external_ip'] = ToIPv4.transform(ip4_val) if ip4_val is not None else None
+    return common_fields
+
+
+def create_regmod_obj(event_dict, event_type, cbr_event):
```
Comment (aviv1ron1) [inline stix_shifter_modules/carbonblack/stix_transmission/event_parser.py:77]: comment that these mappings are to support the same mapping of event fields as cbcloud

File: stix_shifter_modules/carbonblack/stix_translation/results_translator.py (hunk 1 above)
Comment (barvhaim) [reply stix_shifter_modules/carbonblack/stix_translation/results_translator.py:14]: aligned both configuration process's part

**Pull request closed without merging**